witness/src/Dscp/Witness/Logic/Traversals.hs

```diff
@@ -136,6 +137,32 @@ getBlocksFromTo (headerHash -> olderH) (headerHash -> newerH) = runExceptT $ do
 
         pure blocks
 
+-- | Retrieves the requested amount of blocks skipping `skip` blocks from tip.
+getBlocksFrom ::
+       HasWitnessConfig
+    => Word64
+    -> Int
+    -> SdM (Either Text (OldestFirst NonEmpty Block))
+getBlocksFrom skip depthDiff = runExceptT $ do
+    tipIdx <- unDifficulty . hDifficulty <$> lift getTipHeader
+
+    newerH <- ExceptT $ maybeToRight "Can't skip so many blocks" <$> SD.queryOne (Difficulty $ tipIdx - skip)
+
+    let loadCond _block depth = depth < depthDiff
+    blocks <- lift $ loadBlocksDownWhile newerH loadCond
+
+    let retrievedOldest = NE.head (unOldestFirst blocks)
```

Comment:
`head` and `length` from Universum should work as well

Reply:
this is mindless copy-pasta